tests/test_utils.py

```diff
@@ -102,13 +102,20 @@ def test_download_check_etag():
 
     responses.add(responses.GET, url, status=200, body="prova")
 
-    assert utils.download_check_etag(url, "prova.txt")
+    assert utils.download_check_etag(url)
 
     assert os.path.exists("prova.txt")
 
     with open("prova.txt", "r") as f:
         assert f.read() == "prova"
 
+    assert utils.download_check_etag(url, "data/prova.txt")
+
+    assert os.path.exists("data/prova.txt")
```

Comment:
Check that `prova.txt` is not created (I suggest changing data/prova.txt to data/prova2.txt to avoid clashes with the other file in the same test).

Reply:
So basically change data/prova.txt to data/prova2.txt, and assert that data/prova2.txt exists but prova2.txt doesn't.

Reply:
done...:)

Reply:
You forgot to assert that prova2.txt doesn't exist

Reply:
my bad sry ..:)